### PR TITLE
remove TrafficContextHolder.java use sql session to replace

### PR DESCRIPTION
**Pull request by tuichenchuxin**
For #20005.

Changes proposed in this pull request:
- remove TrafficContextHolder.java

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/session/SQLSession.java [hunk 1]
```diff
@@ -15,28 +15,16 @@
...
     
-    @Test
-    public void assertTrafficContextHolder() {
-        assertFalse(TrafficContextHolder.get().isPresent());
-        TrafficContextHolder.set(trafficContext);
-        assertTrue(TrafficContextHolder.get().isPresent());
-        TrafficContextHolder.remove();
-        assertFalse(TrafficContextHolder.get().isPresent());
-    }
+    private final TrafficSessionContext trafficSessionContext = new TrafficSessionContext();
```
[the remark this answers is not in the document]
Comment (tuichenchuxin) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/session/SQLSession.java]: Thanks.Done.

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java [hunk 2]
```diff
@@ -185,19 +184,22 @@ private static SQLFederationDeciderContext decide(final LogicSQL logicSQL, final
...
+    private Optional<String> getInstanceIdAndSave(final LogicSQL logicSQL) {
+        Optional<String> trafficInstanceId = connection.getSqlSession().getTrafficInstanceId();
+        if (!trafficInstanceId.isPresent()) {
+            trafficInstanceId = getTrafficInstanceId(logicSQL);
         }
-        return result;
+        if (connection.isHoldTransaction() && trafficInstanceId.isPresent()) {
+            connection.getSqlSession().setTrafficInstanceId(trafficInstanceId.get());
+        }
+        return trafficInstanceId;
```
Comment (strongduanmu) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:195]: Please rename trafficInstanceId to result.

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java [hunk 3]
```diff
@@ -157,9 +156,9 @@ public ResultSet executeQuery(final String sql) throws SQLException {
         ResultSet result;
         try {
             LogicSQL logicSQL = createLogicSQL(sql);
-            trafficContext = getTrafficContext(logicSQL);
-            if (trafficContext.isMatchTraffic()) {
-                JDBCExecutionUnit executionUnit = createTrafficExecutionUnit(trafficContext, logicSQL);
+            trafficInstanceId = getInstanceIdAndSave(logicSQL).orElse(null);
```
Comment (strongduanmu) [inline shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java:159]: getInstanceIdAndSet may be better.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java]: done.Thanks

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**